### DISCUSSION TITLE
通过增强日志功能改善开发者体验

### DISCUSSION THREAD

**Discussion by hajunho** [PaddlePaddle/PaddleOCR#12006] [Ideas & Features]
大家好，

我对这个项目的潜力印象深刻。我特别期待它未来指向性的功能和结构。但目前，开发者可获取的日志信息非常有限，我想对此提出一些建议。

提案背景
尽管终端用户偏好简洁明了的用户体验，开发者却希望能更详细地理解系统的每个操作过程。当前日志的缺乏可能会在调试和错误处理过程中造成困难。

具体建议
在抽象层上插入日志：在抽象层的关键操作开始和结束时插入日志，帮助开发者更好地理解系统流程。这样做不会影响底层源代码，同时能有效提供信息。
加强模块日志：加强各个模块的关键函数和API调用的日志记录，便于追踪错误和监控性能。
预期效果
提高开发者的调试效率：能够更快更准确地诊断系统内部出现的问题。
改善项目维护：通过准确的日志数据提供，简化项目的维护和管理。
改进其他错误处理：日志有助于更容易地发现和解决意外错误。

`

    def ocr(self, img, det=True, rec=True, cls=True, bin=False, inv=False, alpha_color=(255, 255, 255)):
        assert isinstance(img, (np.ndarray, list, str, bytes))
        if isinstance(img, list) and det == True:
            logger.error('当输入一组图像时，det必须为False')
            exit(0)
        if cls == True and self.use_angle_cls == False:
            logger.warning(
                '由于未初始化角度分类器，因此在前向过程中不会使用它'
            )

        logger.info('开始图像处理...')  # 添加日志
        img = check_img(img, alpha_color)
    
        logger.info('图像处理完成。')  # 添加日志
    
        if isinstance(img, list):
            if self.page_num > len(img) or self.page_num == 0:
                imgs = img
            else:
                imgs = img[:self.page_num]
        else:
            imgs = [img]
    
        def preprocess_image(_image):
            _image = alpha_to_color(_image, alpha_color)
            if inv:
                _image = cv2.bitwise_not(_image)
            if bin:
                _image = binarize_img(_image)
            return _image
    
        logger.info('开始OCR过程...')  # 添加日志
    
        if det and rec:
            ocr_res = []
            for idx, img in enumerate(imgs):
                logger.info(f'正在处理图像 {idx + 1}/{len(imgs)}')  # 添加日志
                img = preprocess_image(img)
                dt_boxes, rec_res, _ = self.__call__(img, cls)
                if not dt_boxes and not rec_res:
                    ocr_res.append(None)
                    continue
                tmp_res = [[box.tolist(), res]
                           for box, res in zip(dt_boxes, rec_res)]
                ocr_res.append(tmp_res)
            return ocr_res
    
        elif det and not rec:
            ocr_res = []
            for idx, img in enumerate(imgs):
                logger.info(f'正在处理图像 {idx + 1}/{len(imgs)}')  # 添加日志
                img = preprocess_image(img)
                dt_boxes, elapse = self.text_detector(img)
                if not dt_boxes:
                    ocr_res.append(None)
                    continue
                tmp_res = [box.tolist() for box in dt_boxes]
                ocr_res.append(tmp_res)
            return ocr_res
    
        else:
            ocr_res = []
            cls_res = []
            for idx, img in enumerate(imgs):
                logger.info(f'正在处理图像 {idx + 1}/{len(imgs)}')  # 添加日志
                if not isinstance(img, list):
                    img = preprocess_image(img)
                    img = [img]
                if self.use_angle_cls and cls:
                    img, cls_res_tmp, elapse = self.text_classifier(img)
                    if not rec:
                        cls_res.append(cls_res_tmp)
                rec_res, elapse = self.text_recognizer(img)
                ocr_res.append(rec_res)
            if not rec:
                return cls_res
    
            logger.info('OCR过程完成。')  # 添加日志
            return ocr_res
`

**Comment by SWHL** [comment]
如果方便的话，欢迎您PR。
